collection-scripts/gather_monitoring

```diff
@@ -1,29 +1,259 @@
+
+MONIT_NS="openshift-monitoring"
+MONIT_ENV="must-gather-env"
+
+JQ_PATH=$(command -v jq 2>/dev/null)
+CURL_CMD=(curl --fail -sk)
+
+BASE_COLLECTION_PATH="/must-gather"
 MONITORING_PATH="${BASE_COLLECTION_PATH}/monitoring"
+PROM_PATH="${MONITORING_PATH}/prometheus"
+GF_PATH="${MONITORING_PATH}/grafana"
+
+test -d "${PROM_PATH}" || mkdir -p "${PROM_PATH}"
+test -d "${GF_PATH}" || mkdir -p "${GF_PATH}"
 
-mkdir -p "${MONITORING_PATH}"
+echo "INFO: Get custom metrics from ConfigMap ${MONIT_ENV} on project ${MONIT_NS}"
+oc -n "${MONIT_NS}" \
+    get cm "${MONIT_ENV}" \
+    -o jsonpath='{.data.env}' > "${MONITORING_PATH}/env" 2> /dev/null
+if [[ $? -eq 0 ]]; then
+    echo "INFO: Loading custom environments variables from ${MONITORING_PATH}/env"
+    source "${MONITORING_PATH}/env"
+else
+    echo "INFO: Unable to load custom environments from ConfigMap, ignoring."
+fi
 
-MONITORING_ROUTE="$(oc get routes -n openshift-monitoring prometheus-k8s -o jsonpath={.status.ingress[0].host})"
+#Session token, overwriten by GAHTER_MONIT_TOKEN
+OCP_TOKEN="${GAHTER_MONIT_TOKEN:-$(oc whoami -t)}"
 SA_TOKEN="$(oc sa get-token default)"
 
-# this is a CA bundle we need to verify the monitoring route, we will write it to disk so we can use it in the flag
-oc -n openshift-config-managed get cm default-ingress-cert -o jsonpath='{.data.ca-bundle\.crt}' > "${MONITORING_PATH}/ca-bundle.crt"
+# this is a CA bundle we need to verify the routes
+oc -n openshift-config-managed \
+    get cm default-ingress-cert \
+    -o jsonpath='{.data.ca-bundle\.crt}' > "${MONITORING_PATH}/ca-bundle.crt"
+
+# Timestamp to Prometheus query time range
+_START_HUNAN=${GAHTER_MONIT_START_DATE:-${GAHTER_MONIT_START_DATE_DEFAULT}}
+DATE_START=$(date -d "${_START}" +%s)
+_END_HUMAN=${GAHTER_MONIT_END_DATE:-${GAHTER_MONIT_END_DATE_DEFAULT}}
+DATE_END=$(date -d "${_END_HUMAN}" +%s)
+
+# Metric resolution: If the resolution is too low, the API will deny due to 11k limitation
+QUERY_STEP=${GAHTER_MONIT_QUERY_STEP:-${GAHTER_MONIT_QUERY_STEP_DEFAULT}}
+
+PROM_HOST=$(oc -n "${MONIT_NS}" get route prometheus-k8s -o jsonpath='{.spec.host}{"\n"}')
+PROM_URL="https://${PROM_HOST}"
+
+GF_HOST=$(oc -n "${MONIT_NS}" get route grafana -o jsonpath='{.spec.host}{"\n"}')
+GF_URL="https://${GF_HOST}"
+
+echo "INFO: Metrics config time range from=${DATE_START} to=${DATE_END} step=${QUERY_STEP}"
+echo "INFO: Config Env GAHTER_MONIT_START_DATE: ${GAHTER_MONIT_START_DATE}"
+echo "INFO: Config Env GAHTER_MONIT_END_DATE: ${GAHTER_MONIT_END_DATE}"
+echo "INFO: Config Env GAHTER_MONIT_QUERY_STEP: ${GAHTER_MONIT_QUERY_STEP}"
+echo "INFO: Config Env GATHER_MONIT_CUSTOM_METRICS: ${GATHER_MONIT_CUSTOM_METRICS}"
+echo "INFO: Config Env GATHER_MONIT_DISCOVERY_PREFIXES: ${GATHER_MONIT_DISCOVERY_PREFIXES}"
+echo "INFO: Config Env GATHER_MONIT_DISCOVERY_GRAFANA: ${GATHER_MONIT_DISCOVERY_GRAFANA}"
+echo "INFO: Prometheus endpoint URL: ${PROM_URL}"
+echo "INFO: Grafana endpoint URL: ${GF_URL}"
+
+function install_jq() {
```

Comment:
This may fail if the cluster is disconnected / behind a proxy that blocks github.com etc ..